### DISCUSSION TITLE
serializeToMiraiCode 没有获取到 MessageSource 的 Mirai 码

### DISCUSSION THREAD

**Discussion by MengYan89** [mamoe/mirai#983] [Q&A] [answered]
使用的版本为2.3.2
根据开发文档
https://github.com/mamoe/mirai/blob/dev/docs/Messages.md#%E6%B6%88%E6%81%AF%E5%85%83%E7%B4%A0
的描述。Message调用serializeToMiraiCode()方法时会获得对应的Mirai码。
而实际调用返回的效果却与contentToSting()相同。
代码如下
```
            bot.getEventChannel().subscribeAlways(GroupMessageEvent.class, event -> {
                System.out.println(event.getGroup().getId()+":"+event.getMessage());
                System.out.println(event.getMessage().contentToString());
                System.out.println(event.getMessage().serializeToMiraiCode());
            });
```
控制台输出结果分别为
```
910726832:[mirai:source:[513],[-348440005]]555
555
555
```

**Comment by Him188** [comment]
> 控制台输出结果分别为

~是不是少了一行?~ 发现是 markdown 格式问题

**Comment by Him188** [comment, marked as answer]
只有 [表中列出](https://github.com/mamoe/mirai/blob/dev/docs/Messages.md#%E7%94%B1-codablemessage-%E5%8F%96%E5%BE%97-mirai-%E7%A0%81%E5%AD%97%E7%AC%A6%E4%B8%B2) 的类型支持 Mirai 码, 不支持的类型会以空字符串表示

**Comment by MengYan89** [comment]
> 只有[表中列出](https://github.com/mamoe/mirai/blob/dev/docs/Messages.md#%E7%94%B1-codablemessage-%E5%8F%96%E5%BE%97-mirai-%E7%A0%81%E5%AD%97%E7%AC%A6%E4%B8%B2)的类型支持Mirai码，不支持的类型会以空字符串表示

这里event.getMessage获取到的类型是MessageChain根据文档的描述会是多个mirai 码相连，不应该会出现这种调用后直接显示消息原本内容才对

**Comment by Him188** [comment]
> > 只有[表中列出](https://github.com/mamoe/mirai/blob/dev/docs/Messages.md#%E7%94%B1-codablemessage-%E5%8F%96%E5%BE%97-mirai-%E7%A0%81%E5%AD%97%E7%AC%A6%E4%B8%B2)的类型支持Mirai码，不支持的类型会以空字符串表示
> 
> 这里event.getMessage获取到的类型是MessageChain根据文档的描述会是多个mirai 码相连，不应该会出现这种调用后直接显示消息原本内容才对

MessageSource 不支持 Mirai 码, 因此是空字符串
